File: August 2025/05.cpp
class Solution {
  public:
    bool isPalinSent(string &s) {
        string cleaned;
        
        // Filter out non-alphanumeric and convert to lowercase
        for (char ch : s) {
            if (isalnum(ch)) {
                cleaned += tolower(ch);
            }
        }

        // Check if the cleaned string is a palindrome
        int left = 0, right = cleaned.size() - 1;
        while (left < right) {
            if (cleaned[left] != cleaned[right]) {
                return false;
            }
            ++left;
            --right;
        }

        return true;
    }
};

File: August 2025/05.py
class Solution:
    def isPalinSent(self, s):
        # Filter out non-alphanumeric characters and convert to lowercase
        cleaned = ''.join(ch.lower() for ch in s if ch.isalnum())
        # Check if the cleaned string is the same forwards and backwards
        return cleaned == cleaned[::-1]
